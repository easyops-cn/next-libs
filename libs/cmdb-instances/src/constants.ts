export const UploadFileUrl =
  "api/gateway/object_store.object_store.PutObject/api/v1/objectStore/bucket/cmdb-bucket/object";

export const DownloadFileUrl =
  "api/gateway/object_store.object_store.GetObject/api/v1/objectStore/bucket/cmdb-bucket/object";
