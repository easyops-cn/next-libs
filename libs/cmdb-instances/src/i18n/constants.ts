export const NS_LIBS_CMDB_INSTANCES = "libs-cmdb-instances";

export enum K {
  VALIDATE_MESSAGE_REQUIRED = "VALIDATE_MESSAGE_REQUIRED",
  DYNAMIC_FILTER = "DYNAMIC_FILTER",
  TIP = "TIP",
  DELETE = "DELETE",
  INVALID_OR_FORBIDDEN_IPS = "INVALID_OR_FORBIDDEN_IPS",
  INVALID = "INVALID",
  SELECT_FROM_CMDB = "SELECT_FROM_CMDB",
  IP_PLACEHOLDER = "IP_PLACEHOLDER",
  CLICK_TO_SELECT = "CLICK_TO_SELECT",
  MATCHING_REGULAR = "MATCHING_REGULAR",
  ADD = "ADD",
  ENTER_MULTIPLE_STRING_WITH_ENTER_KEY_AS_THE_SEPARATOR = "ENTER_MULTIPLE_STRING_WITH_ENTER_KEY_AS_THE_SEPARATOR",
  UNIQUE_NO_REPEAT = "UNIQUE_NO_REPEAT",
  TYPE_NO_SUPPORT_EDIT = "TYPE_NO_SUPPORT_EDIT",
  TEMPORARILY_NOT_CHOOSE = "TEMPORARILY_NOT_CHOOSE",
  ENUM_ERROR_TIP = "ENUM_ERROR_TIP",
  STRUCT_ERROR_TIP = "STRUCT_ERROR_TIP",
  LINK = "LINK",
  LINK_PLACEHOLDER = "LINK_PLACEHOLDER",
  TITLE = "TITLE",
  TITLE_PLACEHOLDER = "TITLE_PLACEHOLDER",
  CLICK_TO_FILTER = "CLICK_TO_FILTER",
  CLICK_TO_CANCEL_FILTER = "CLICK_TO_CANCEL_FILTER",
  CLICK_TO_HIDDEN = "CLICK_TO_HIDDEN",
  CLICK_TO_SHOW = "CLICK_TO_SHOW",
  RELATED_TO_ME = "RELATED_TO_ME",
  DISPLAY_OMITTED_INFORMATION = "DISPLAY_OMITTED_INFORMATION",
  CLEAR = "CLEAR",
  NORMAL_HOST = "NORMAL_HOST",
  CURRENT_FILTER_REQUIREMENTS = "CURRENT_FILTER_REQUIREMENTS",
  ADVANCED_SEARCH = "ADVANCED_SEARCH",
  CHOSEN_OPTIONS = "CHOSEN_OPTIONS",
  FUZZY_SEARCH = "FUZZY_SEARCH",
  BASIC_INFORMATION = "BASIC_INFORMATION",
  MORE = "MORE",
  DEFAULT_ATTRIBUTE = "DEFAULT_ATTRIBUTE",
  CANCEL = "CANCEL",
  CLOSE = "CLOSE",
  DISPLAY_SETTINGS = "DISPLAY_SETTINGS",
  FIELD_SETTINGS = "FIELD_SETTINGS",
  SEARCH_BY_FIELD_NAME = "SEARCH_BY_FIELD_NAME",
  SELECT_ALL = "SELECT_ALL",
  RESTORE_DEFAULT = "RESTORE_DEFAULT",
  SET_DEFAULT_DISPLAY_SUCCESS = "SET_DEFAULT_DISPLAY_SUCCESS",
  DEFAULT_DISPLAY = "DEFAULT_DISPLAY",
  CONFIRM = "CONFIRM",
  ONLY_ONE_INSTANCE_TO_ALLOWED = "ONLY_ONE_INSTANCE_TO_ALLOWED",
  OPERATOR_CONTAIN_DEFINE = "OPERATOR_CONTAIN_DEFINE",
  OPERATOR_NOT_CONTAIN_DEFINE = "OPERATOR_NOT_CONTAIN_DEFINE",
  OPERATOR_EQUAL_DEFINE = "OPERATOR_EQUAL_DEFINE",
  OPERATOR_NOT_EQUAL_DEFINE = "OPERATOR_NOT_EQUAL_DEFINE",
  OPERATOR_IS_EMPTY_DEFINE = "OPERATOR_IS_EMPTY_DEFINE",
  OPERATOR_IS_NOT_EMPTY_DEFINE = "OPERATOR_IS_NOT_EMPTY_DEFINE",
  OPERATOR_BETWEEN_DEFINE_TEXT = "OPERATOR_BETWEEN_DEFINE_TEXT",
  OPERATOR_BETWEEN_DEFINE_TEXT_TIME = "OPERATOR_BETWEEN_DEFINE_TEXT_TIME",
  OPERATOR_BETWEEN_DEFINE_TEXT_INT = "OPERATOR_BETWEEN_DEFINE_TEXT_INT",
  OPERATOR_BETWEEN_DEFINE_TEXT_THE = "OPERATOR_BETWEEN_DEFINE_TEXT_THE",
  SEARCH = "SEARCH",
  RESET = "RESET",
  CREATOR = "CREATOR",
  CREATION_TIME = "CREATION_TIME",
  MODIFIER = "MODIFIER",
  LAST_MODIFICATION_TIME = "LAST_MODIFICATION_TIME",
  DEVELOPMENT = "DEVELOPMENT",
  VIEW = "VIEW",
  ITEM_ADD_IN_BATCHES = "ITEM_ADD_IN_BATCHES",
  SAVE = "SAVE",
  SAVE_AND_CONTINUE = "SAVE_AND_CONTINUE",
  MODIFICATION = "MODIFICATION",
  NOT_MEET_REGEX = "NOT_MEET_REGEX",
  NOT_MEET_REGEX_DETAIL = "NOT_MEET_REGEX_DETAIL",
  NOT_MEET_JSON = "NOT_MEET_JSON",
  NOT_MEET_XML = "NOT_MEET_XML",
  CREATE_ANOTHER = "CREATE_ANOTHER",
  DELETE_STRUCT_CONFIRM_MSG = "DELETE_STRUCT_CONFIRM_MSG",
  DELETE_INSTANCE_CONFIRM_MSG = "DELETE_INSTANCE_CONFIRM_MSG",
  OPERATION = "OPERATION",
  VIEW_ALL_SELECTED_INSTANCES = "VIEW_ALL_SELECTED_INSTANCES",
  CHOOSE_INSTANCE = "CHOOSE_INSTANCE",
  CHOOSE_INSTANCE_NAME = "CHOOSE_INSTANCE_NAME",
  VIEW_ALL_DATA = "VIEW_ALL_DATA",
  FILTER_FROM_CMDB = "FILTER_FROM_CMDB",
  PAGINATION_TOTAL_TEXT = "PAGINATION_TOTAL_TEXT",
  PAGINATION_TOTAL_UNIT = "PAGINATION_TOTAL_UNIT",
  JUMP_TO = "JUMP_TO",
  INSTANCE_DETAIL = "INSTANCE_DETAIL",
  COPY = "COPY",
  COPY_SUCCESS = "COPY_SUCCESS",
  // FILTER_FROM_CMDB = "FILTER_FROM_CMDB",
  USERS = "USERS",
  USER_GROUPS = "USER_GROUPS",
  SWITCH = "SWITCH",
  USERS_RESULT_LABEL = "USERS_RESULT_LABEL",
  NO_DATA = "NO_DATA",
  USER_GROUPS_RESULT_LABEL = "USER_GROUPS_RESULT_LABEL",
  PERMISSION_WHITELIST = "PERMISSION_WHITELIST",
  FILTER_FROM_CMDB_TYPE = "FILTER_FROM_CMDB_TYPE",
  ALL_CLUSTER = "ALL_CLUSTER",
  FREE_SELECTION = "FREE_SELECTION",
  APP_SELECTION = "APP_SELECTION",
  ATTRIBUTE_NAME_REQUIRED = "ATTRIBUTE_NAME_REQUIRED",
  INSTANCE_SOURCE = "INSTANCE_SOURCE",
  INSTANCE_SOURCE_TOOLTIP = "INSTANCE_SOURCE_TOOLTIP",
  INSTANCE_SOURCE_TAG_TEXT = "INSTANCE_SOURCE_TAG_TEXT",
  COPY_SELECTED_IP = "COPY_SELECTED_IP",
  SELECT_COPY_DATA = "SELECT_COPY_DATA",
  ADVANCE_SEARCH_SINGLE_INPUT_PLACEHOLDER = "ADVANCE_SEARCH_SINGLE_INPUT_PLACEHOLDER",
  NUMBER_INPUT_PLACEHOLDER = "NUMBER_INPUT_PLACEHOLDER",
  FIX_HEADER = "FIX_HEADER",
  CANCEL_FIX_HEADER = "CANCEL_FIX_HEADER",
  VIEW_SPECIFIC_INSTANCES = "VIEW_SPECIFIC_INSTANCES",
  VIEW_MORE = "VIEW_MORE",
  CREATE_AND_BIND = "CREATE_AND_BIND",
  PERMISSION_DENIED = "PERMISSION_DENIED",
  CLICK_TO_FILTER_INSTANCE_SOURCE = "CLICK_TO_FILTER_INSTANCE_SOURCE",
  SEARCH_MODEL = "SEARCH_MODEL",
  UPLOAD_SIZE_LARGE_TEXT = "UPLOAD_SIZE_LARGE_TEXT",
  UPLOAD = "UPLOAD",
  BACKGROUND_SEARCH = "BACKGROUND_SEARCH",
  BACK = "BACK",
  RELATION_INSTANCE_TOOLTIP = "RELATION_INSTANCE_TOOLTIP",
}

export type Locale = { [key in K]: string };
