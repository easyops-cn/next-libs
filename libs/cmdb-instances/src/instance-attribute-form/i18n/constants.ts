export const NS_CMDB_INSTANCES = "libs-cmdb-instances";

export enum K {
  VALIDATE_MESSAGE_REQUIRED = "VALIDATE_MESSAGE_REQUIRED"
}

export type Locale = { [key in K]: string };
