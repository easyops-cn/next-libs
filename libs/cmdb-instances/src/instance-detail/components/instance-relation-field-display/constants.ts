export const userDisplayFieldIdList = [
  "name",
  "nickname",
  "user_email",
  "user_tel",
  "user_memo",
  "state",
];
