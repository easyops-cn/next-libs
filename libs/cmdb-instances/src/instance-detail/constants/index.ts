export const DEFAULT_ATTRIBUTE_TAG = "基本信息";
export const BASIC_INFORMATION_RELATION_GROUP_ID = "basic_info";
export const DEFAULT_ATTRIBUTE_TAG_STR = "默认属性";
export const BASIC_INFO = "基本信息";
