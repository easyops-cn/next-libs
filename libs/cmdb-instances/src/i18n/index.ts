import i18next from "i18next";
import { NS_LIBS_CMDB_INSTANCES } from "./constants";
import en from "./locales/en";
import zh from "./locales/zh";

// istanbul ignore next
export function addResourceBundle(): void {
  i18next.addResourceBundle("en", NS_LIBS_CMDB_INSTANCES, en);
  i18next.addResourceBundle("zh", NS_LIBS_CMDB_INSTANCES, zh);
}
