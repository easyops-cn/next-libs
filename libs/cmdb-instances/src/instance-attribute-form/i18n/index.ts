import i18next from "i18next";
import { NS_CMDB_INSTANCES } from "./constants";
import en from "./locales/en";
import zh from "./locales/zh";

export function addResourceBundle(): void {
  i18next.addResourceBundle("en", NS_CMDB_INSTANCES, en);
  i18next.addResourceBundle("zh", NS_CMDB_INSTANCES, zh);
}
