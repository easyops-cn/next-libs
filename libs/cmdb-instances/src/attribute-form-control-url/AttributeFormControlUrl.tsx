import { FormControl } from "../model-attribute-form-control/ModelAttributeFormControl";
import { Input } from "antd";
import i18n from "i18next";
import { NS_LIBS_CMDB_INSTANCES, K } from "../i18n/constants";
import React, { Component, ReactNode } from "react";
import "./AttributeFormControlUrl.module.css";

export interface AttributeFormControlUrlProps extends Partial<FormControl> {
  value: string;
  onChange: (value: string) => void;
}

export class AttributeFormControlUrl extends Component<AttributeFormControlUrlProps> {
  private url: string;
  private title: string;

  constructor(props: AttributeFormControlUrlProps) {
    super(props);

    const { title, url } = AttributeFormControlUrl.breakDownValue(props.value);
    this.title = title;
    this.url = url;
  }

  static breakDownValue(value: string): { title: string; url: string } {
    const pattern = /^\[(.*)\]\((.*)\)$/;
    const arr = value && value.match(pattern);
    if (arr) {
      return {
        title: arr[1],
        url: AttributeFormControlUrl.getFormattedUrl(arr[2]),
      };
    }

    return {
      title: value,
      url: value,
    };
  }

  static getFormattedUrl(url: string): string {
    /* istanbul ignore if  */
    if (url) {
      const pattern = /^(http|https):\/\/.*$/;
      // return pattern.test(url) ? url : "http://" + url;
      const urlReg = /^((http|https):\/\/)?(([A-Za-z0-9]+-[A-Za-z0-9]+|[A-Za-z0-9]+)\.)+([A-Za-z]+)[/\?\:]?.*|^((2[0-4]\d|25[0-5]|[01]?\d\d?)\.){3}(2[0-4]\d|25[0-5]|[01]?\d\d?)$/;
      const reg = /^\//
      const origin = window.location.origin;
      return pattern.test(url) ? url : urlReg.test(url) ? url : `${origin}${reg.test(url) ? url : '/' + url} `;
    }
    return "";
  }

  // preSetValidation = () => {
  //   const url = AttributeFormControlUrl.getFormattedUrl(this.url);

  //   if (this.url !== url) {
  //     this.url = url;
  //     this.composeUrl();
  //   }
  // };

  /* istanbul ignore next */
  composeUrl(): void {
    const title = this.title ?? "";
    const url = AttributeFormControlUrl.getFormattedUrl(this.url);
    const value = url || title ? `[${title}](${url})` : "";
    this.props.onChange(value);
  }

  handleUrlChange = (url: string) => {
    this.url = url;
    this.composeUrl();
  };

  handleTitleChange = (title: string) => {
    this.title = title;
    this.composeUrl();
  };

  render(): ReactNode {
    const { readOnly, placeholder } = this.props;
    const { title, url } = this;

    return (
      <div>
        <div className="singleContainer">
          <label className="presetTitle">
            {i18n.t(`${NS_LIBS_CMDB_INSTANCES}:${K.LINK}`)}
          </label>
          <Input
            readOnly={readOnly}
            value={url}
            type="url"
            onChange={(e) => this.handleUrlChange(e.target.value)}
            // onBlur={this.preSetValidation}
            placeholder={
              placeholder ||
              i18n.t(`${NS_LIBS_CMDB_INSTANCES}:${K.LINK_PLACEHOLDER}`)
            }
          />
        </div>

        <div className="singleContainer">
          <label className="presetTitle">
            {i18n.t(`${NS_LIBS_CMDB_INSTANCES}:${K.TITLE}`)}
          </label>
          <Input
            defaultValue={title}
            type="text"
            onChange={(e) => this.handleTitleChange(e.target.value)}
            // onBlur={this.preSetValidation}
            placeholder={i18n.t(
              `${NS_LIBS_CMDB_INSTANCES}:${K.TITLE_PLACEHOLDER}`
            )}
          />
        </div>
      </div>
    );
  }
}
