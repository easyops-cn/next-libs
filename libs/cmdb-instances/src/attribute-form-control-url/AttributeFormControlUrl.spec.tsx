import React from "react";
import { mount, ReactWrapper } from "enzyme";
import { AttributeFormControlUrl } from "./AttributeFormControlUrl";
import i18n from "i18next";
import { K, NS_LIBS_CMDB_INSTANCES } from "../i18n/constants";

describe("AttributeFormControlUrl", () => {
  let wrapper: ReactWrapper<
    AttributeFormControlUrl["props"],
    AttributeFormControlUrl["state"],
    React.Component<Record<any, any>, Record<any, any>, any>
  >;
  let instance: React.Component<Record<any, any>, Record<any, any>, any>;
  const props = {
    value: "[百度](http://www.baidu.com)",
    onChange: jest.fn(),
    readOnly: true,
  };

  beforeEach(() => {
    wrapper = mount(<AttributeFormControlUrl {...props} />);
    instance = wrapper.instance();
  });

  afterEach(() => {
    props.onChange.mockClear();
  });

  it("should change url value", () => {
    const urlEle = wrapper.find("input").at(0);

    urlEle.simulate("change", {
      target: {
        value: "www.google.com",
      },
    });

    expect(props.onChange).toHaveBeenCalledWith("[百度](www.google.com)");

    urlEle.simulate("blur");
    expect(props.onChange).toHaveBeenCalledWith(
      "[百度](www.google.com)"
    );
    urlEle.simulate("change", {
      target: {
        value: "/google.com",
      },
    });
    urlEle.simulate("blur");
    expect(props.onChange).toHaveBeenCalledWith(
      "[百度](www.google.com)"
    );
    urlEle.simulate("change", {
      target: {
        value: null,
      },
    });
    urlEle.simulate("blur");
    expect(props.onChange).toHaveBeenCalledWith(
      "[百度](www.google.com)"
    );
  });

  it("should change title value", () => {
    const titleEle = wrapper.find("input").at(1);

    titleEle.simulate("change", {
      target: {
        value: "google",
      },
    });

    expect(props.onChange).toHaveBeenCalledWith(
      "[google](http://www.baidu.com)"
    );

    titleEle.simulate("blur");
    expect(props.onChange).toHaveBeenCalledTimes(1);
  });

  it("value should be null when props.value to equal null", () => {
    const newProps = Object.assign({}, props, {
      value: null,
    });
    const wrapper = mount(<AttributeFormControlUrl {...newProps} />);

    const [urlEle, titleEle] = wrapper.find("input").map((node) => node);
    expect(urlEle.props().value).toBeFalsy();
    expect(titleEle.props().value).toBeFalsy();
  });

  it("should work", () => {
    const { title, url } = AttributeFormControlUrl.breakDownValue(props.value);
    const [urlEle, titleEle] = wrapper.find("input").map((node) => node);

    expect(urlEle.exists() && titleEle.exists()).toBeTruthy();
    expect(urlEle.props().value).toEqual(url);
    // expect(urlEle.props().placeholder).toBe("请输入链接");
    expect(urlEle.props().placeholder).toBe(
      i18n.t(`${NS_LIBS_CMDB_INSTANCES}:${K.LINK_PLACEHOLDER}`)
    );
    expect(urlEle.props().type).toBe("url");
    expect(urlEle.props().readOnly).toBeTruthy();

    expect(titleEle.props().value).toEqual(title);
    // expect(titleEle.props().placeholder).toBe("选填，请输入显示的标题");
    expect(titleEle.props().placeholder).toBe(
      i18n.t(`${NS_LIBS_CMDB_INSTANCES}:${K.TITLE_PLACEHOLDER}`)
    );
    expect(titleEle.props().type).toBe("text");

    expect(wrapper).toMatchSnapshot();
  });
});
