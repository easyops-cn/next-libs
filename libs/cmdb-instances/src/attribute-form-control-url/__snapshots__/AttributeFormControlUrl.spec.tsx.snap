// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributeFormControlUrl should work 1`] = `
<AttributeFormControlUrl
  onChange={[MockFunction]}
  readOnly={true}
  value="[百度](http://www.baidu.com)"
>
  <div>
    <div
      className="singleContainer"
    >
      <label
        className="presetTitle"
      >
        libs-cmdb-instances:LINK
      </label>
      <Input
        onChange={[Function]}
        placeholder="libs-cmdb-instances:LINK_PLACEHOLDER"
        readOnly={true}
        type="url"
        value="http://www.baidu.com"
      >
        <ClearableLabeledInput
          bordered={true}
          element={
            <input
              className="ant-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="libs-cmdb-instances:LINK_PLACEHOLDER"
              readOnly={true}
              type="url"
              value="http://www.baidu.com"
            />
          }
          focused={false}
          handleReset={[Function]}
          inputType="input"
          onChange={[Function]}
          placeholder="libs-cmdb-instances:LINK_PLACEHOLDER"
          prefixCls="ant-input"
          readOnly={true}
          triggerFocus={[Function]}
          type="url"
          value="http://www.baidu.com"
        >
          <input
            className="ant-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="libs-cmdb-instances:LINK_PLACEHOLDER"
            readOnly={true}
            type="url"
            value="http://www.baidu.com"
          />
        </ClearableLabeledInput>
      </Input>
    </div>
    <div
      className="singleContainer"
    >
      <label
        className="presetTitle"
      >
        libs-cmdb-instances:TITLE
      </label>
      <Input
        defaultValue="百度"
        onChange={[Function]}
        placeholder="libs-cmdb-instances:TITLE_PLACEHOLDER"
        type="text"
      >
        <ClearableLabeledInput
          bordered={true}
          defaultValue="百度"
          element={
            <input
              className="ant-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="libs-cmdb-instances:TITLE_PLACEHOLDER"
              type="text"
            />
          }
          focused={false}
          handleReset={[Function]}
          inputType="input"
          onChange={[Function]}
          placeholder="libs-cmdb-instances:TITLE_PLACEHOLDER"
          prefixCls="ant-input"
          triggerFocus={[Function]}
          type="text"
          value="百度"
        >
          <input
            className="ant-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="libs-cmdb-instances:TITLE_PLACEHOLDER"
            type="text"
            value="百度"
          />
        </ClearableLabeledInput>
      </Input>
    </div>
  </div>
</AttributeFormControlUrl>
`;
